pkg/subctl/cmd/utils/restconfig/restconfig.go

```diff
@@ -39,24 +44,85 @@ type RestConfig struct {
 	ClusterName string
 }
 
-func MustGetForClusters(kubeConfigPath string, kubeContexts []string) []RestConfig {
-	configs, err := ForClusters(kubeConfigPath, kubeContexts)
+type Producer struct {
```

Comment:
With @Jaanki efforts to restructure/refactor the code base, perhaps we should move/put this under _pkg/restconfig_ to avoid having to refactor it later.

Reply:
I moved it to `internal/restconfig`.

Reply:
The code references `cobra.Command` which I think seems more suited under _cmd_  but it's fine either way.

Reply:
AIUI, `cmd` is supposed to be kept as small as possible, with all the actual code under `internal` or `pkg`.

Reply:
yeah it's reasonable to keep `cmd` relatively small with the non-CLI related elsewhere so it can be reused otherwise. But the `Producer` is tied to `kubeConfig` and `kubeContext` and references `cobra.Command` which seem pertinent to CLI and leans me towards _cmd/restconfig_.

On a side note, we talked about using this for `deploy.Broker` under _pkg/deploy/broker_  but, to make it generally reusable (eg for the **submariner-addon**) and also "unit-testable", we should be passing client around set interfaces instead.

Reply:
See https://github.com/golang-standards/project-layout#cmd for where this is coming from. `cmd/restconfig` would suggest we’re building a `restconfig` command, which isn’t the case here.

Reply:
I agree. This should not be in `cmd/subctl` but rather `internal/restconfig` seems a good fit.

Reply:
OK.  Do we anticipate external apps using this in which case it should be under _pkg/restconfig_?

Reply:
This should stay confined to `subctl` for now IMO.

Reply:
OK - _internal/restconfig_ it is. We can always move it later if need be.